lib/cdo/aws/s3_packaging.rb

```diff
@@ -0,0 +1,143 @@
+require 'aws-sdk'
+
+class S3Packaging
+  BUCKET_NAME = 'cdo-build-package'
+
+  def initialize(package_name, source_location, target_location)
+    @client = Aws::S3::Client.new
+    @commit_hash = RakeUtils.git_latest_commit_hash source_location
+    @package_name = package_name
+    @source_location = source_location
+    @target_location = target_location
+  end
+
+  # Tries to get an up to date package without building
+  # @return True if our package is now up to date
+  def update_from_s3
+    begin
+      ensure_updated_package
+    rescue Aws::S3::Errors::NoSuchKey
+      puts 'Package does not exist on S3'
+      return false
+    rescue Exception => e
+      puts "update_from_s3 failed: #{e.message}"
+      return false
+    end
+    return true
+  end
+
+  # creates a package from the given assets location and upload it to s3
+  # @return tempfile object of package
+  def upload_packge_to_s3(sub_path)
```

Comment:
typo in `packge`, or intentional (since it's referenced consistently throughout)?